packages/codec/src/blockchain.ts

```diff
@@ -77,3 +79,211 @@ export const WitnessArgs = WitnessArgsOf({
   input_type: HexifyCodec,
   output_type: HexifyCodec,
 });
+
+/**
+ * Implementation of blockchain.mol
+ * https://github.com/nervosnetwork/ckb/blob/5a7efe7a0b720de79ff3761dc6e8424b8d5b22ea/util/types/schemas/blockchain.mol
+ */
+export const HashType = createFixedBytesCodec<string>({
+  byteLength: 1,
+  pack: (type) => {
+    const data = new DataView(new ArrayBuffer(1));
+    if (type === "data") {
+      data.setUint8(0, 0);
+    } else if (type === "type") {
+      data.setUint8(0, 1);
+    } else if (type === "data1") {
+      data.setUint8(0, 2);
+    } else {
+      throw new Error(`invalid hash type: ${type}`);
+    }
+    return new Uint8Array(data.buffer);
+  },
+  unpack: (buf) => {
+    const data = new DataView(buf.buffer).getUint8(0);
+    if (data === 0) {
+      return "data";
+    } else if (data === 1) {
+      return "type";
+    } else if (data === 2) {
+      return "data1";
+    } else {
+      throw new Error("invalid data");
+    }
+  },
+});
```

Comment:
```suggestion
import { HashType as _HashType } from "@ckb-lumos/base";

export const HashType = createFixedBytesCodec<_HashType>({
  byteLength: 1,
  pack: (type) => {
    if (type === "data") return Uint8.pack(0);
    if (type === "type") return Uint8.pack(1);
    if (type === "data1") return Uint8.pack(2);

    throw new Error(`Invalid hash type: ${type}`);
  },
  unpack: (buf) => {
    const hashTypeBuf = Uint8.unpack(buf);
    if (hashTypeBuf === 0) return "data";
    if (hashTypeBuf === 1) return "type";
    if (hashTypeBuf === 2) return "data1";

    throw new Error(`Invalid hash type: ${hashTypeBuf}`);
  },
});
```
we can use the existing codec to simplify the code